### ISSUE TITLE
Scalability

### ISSUE THREAD

**Issue by sffc** [davidben/client-language-selection#2]
Web sites can be translated into dozens or hundreds of languages, and may also include regional variants and script variants.  In your proposal, it looks like the server has to send a full list of supported languages to the client.  How does this scale?

**Comment by davidben** [comment]
This is true of `Accept-Language` being used with HTTP Representation Variants as a whole, but that's certainly a concern. HTTP/2 header compression means you'd only send the list once per connection, and there's a static Huffman encoding to make it a little smaller, but the list may still be a bit large. I think a dozen or two should be fine here, but I agree hundreds is pushing it. :-)

For such sites, I wonder if it makes sense to treat a `Variants`-less `Vary: Accept-Language` (or some other signal) as supporting ~all languages and the client just requests the most preferred one.
